src/java/htsjdk/samtools/reference/IndexedFastaSequenceFile.java

```diff
@@ -202,10 +251,20 @@ public ReferenceSequence getSubsequenceAt( String contig, long start, long stop
             startOffset += Math.max((int)(startOffset%bytesPerLine - basesPerLine + 1),0);
 
             try {
-                 startOffset += channel.read(channelBuffer,indexEntry.getLocation()+startOffset);
+                if (channel instanceof FileChannel) {
+                    startOffset += ((FileChannel) channel).read(channelBuffer,indexEntry.getLocation()+startOffset);
```

Comment:
Is this file channel method more performant than the method in the `else`?  I.e. is there any reason to have 2 paths through here instead of just the 1.  The file channel method is obviously more readable, but if it doesn't offer any other advantage I'd probably go with just the second path through the code. 

If you think it's worth have the `if else` in there to handle the file channel case, could you extract this logic into a method.